creating_geometric_models/create_fault_from_trace.py

```diff
@@ -63,15 +59,17 @@ def generate_vertices(depth, sign=1):
     nd = depth.shape[0]
     vertices = np.zeros((nx, nd, 3))
     vertices[:, 0, :] = nodes
+    if args.dipType == 0:
+        # moved out of the loop: call to math.tan slow down significantly the code
+        one_over_tan_dip = 1.0 / tan(dip)
     for i in range(0, nx):
+        if args.dipType > 1:
+            one_over_tan_dip = 1.0 / tan(aDip[i])
         for j in range(1, nd):
-            if args.dipType == 0:
-                mydip = dip
-            elif args.dipType == 1:
+            if args.dipType == 1:
                 mydip = dipangle(depth[j] + vertices[i, 0, 2])
-            else:
-                mydip = aDip[i]
-            ud = -(1.0 / tan(mydip) * av1[i, :] - uz)
+                one_over_tan_dip = 1.0 / tan(mydip)
+            ud = -(one_over_tan_dip * av1[i, :] - uz)
```

Comment:
If there is a speed problem with this loop, could you use vectorized numpy functions?

Reply:
Actually, I use this python script as a base for another script that adds roughness to the fault.
In this case, we need a very fine discretization, and to run many times the loop.
In such a case, moving the tan outside the loop makes a lot of sense (maybe run time ~10 times larges with tan in the loop).
The loop cannot be vectorized because mydip = dipangle(depth[j] + vertices[i, 0, 2]) depends on vertices[i, 0, 2] which is calculated within the loop.
Let's do further optimization when needed but so far, it works fast enough.